api/baremetal/v1alpha1/server_utils.go

```diff
@@ -92,3 +92,40 @@ func (s *API) WaitForServerInstall(req *WaitForServerInstallRequest) (*Server, e
 
 	return server.(*Server), nil
 }
+
+// getServerOfferName returns the offer name of a baremetal server
+func (s *API) GetServerOfferName(server *Server) (string, error) {
+	offer, err := s.GetOffer(&GetOfferRequest{
+		OfferID: server.OfferID,
+		Zone:    server.Zone,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	return offer.Name, nil
+}
+
+type GetOfferIDFromOfferNameRequest struct {
+	OfferName string
+	Zone      scw.Zone
+	Timeout   time.Duration
+}
+
+func (s *API) GetOfferIDFromName(req *GetOfferIDFromOfferNameRequest) (string, error) {
+	res, err := s.ListOffers(&ListOffersRequest{
+		Zone: req.Zone,
+	}, scw.WithAllPages())
+
```

Comment:
remove newline: we usually group treatment of the error with where it could be generated